azkaban-common/src/main/java/azkaban/utils/StdOutErrRedirect.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.utils;
+
+import java.io.OutputStream;
+import java.io.PrintStream;
+
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+
+/**
+ * A class to encapsulate the redirection of stdout and stderr to log4j
+ * This allows us to catch messages written to the console (although we should
+ * not be using System.out to write out).
+ */
+
+public class StdOutErrRedirect {
+  private static final Logger logger = Logger.getLogger(StdOutErrRedirect.class);
+  private static PrintStream infoStream = createStream(System.out, Level.INFO);
+  private static PrintStream errorStream = createStream(System.out, Level.ERROR);
+
+  public static void redirectOutAndErrToLog() {
+    System.setOut(infoStream);
+    System.setErr(errorStream);
+  }
+
+  private static PrintStream createStream(PrintStream stream, Level level) {
+    return new LogStream(stream, level);
+  };
+
+  private static class LogStream extends PrintStream {
+    private Level level;
+
+    public LogStream(OutputStream out, Level level) {
+      super(out);
+      this.level = level;
+    }
+    // String - all other overridden methods will call this
+    @Override
+    public void println(final String string) {
+      print(string);
+    }
+    @Override
+    public void print(final String string) {
+      logger.log(level, string);
```

Comment:
Can we have a separate differently named private method which does this, similar to the OutputStream in java. That makes the code much more readable than calling an already heavily overloaded and overridden method.